lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/BinarySpec.hs

```diff
@@ -90,8 +102,59 @@ spec = do
         it "throws when length (key) != 32" $
             evaluate (singleAddressFromKey (Proxy @'Mainnet) (XPub "\148" cc))
             `shouldThrow` anyErrorCall
+    describe "keyToAddress is consistent with jcli" $ do
+        -- The goldens have been verified with jcli as follows:
+        -- $ jcli key generate --type=Ed25519 | jcli key to-public
+        -- ed25519_pk1yv0er3wlzvcauqj470tesdlcwy4dll9w7cvsvn4q30678gh44tnsxdh75c
+        -- $ jcli address single ed25519_pk1yv0er3wlzvcauqj470tesdlcwy4dll9w7cvsvn4q30678gh44tnsxdh75c --testing
+        -- ta1sv33lyw9mufnrhsz2hea0xphlpcj4hlu4mmpjpjw5z9ltcaz7k4wwywkd6j
+        -- $ jcli address single ed25519_pk1yv0er3wlzvcauqj470tesdlcwy4dll9w7cvsvn4q30678gh44tnsxdh75c
+        -- ca1qv33lyw9mufnrhsz2hea0xphlpcj4hlu4mmpjpjw5z9ltcaz7k4ww0xfchg
+
+        bech32KeyToAddrGolden
+            "ed25519_pk1yv0er3wlzvcauqj470tesdlcwy4dll9w7cvsvn4q30678gh44tnsxdh75c"
+            "ta1sv33lyw9mufnrhsz2hea0xphlpcj4hlu4mmpjpjw5z9ltcaz7k4wwywkd6j"
+            "ca1qv33lyw9mufnrhsz2hea0xphlpcj4hlu4mmpjpjw5z9ltcaz7k4ww0xfchg"
+
+        bech32KeyToAddrGolden
+            "ed25519_pk1gf85kgfspqzgz9mvmral4e8zatg6d64ep3gk223fla09parjm7zqf4xy3n"
+            "ta1sdpy7jepxqyqfqghdnv0h7hyut4drfh2hyx9zef298l4u585wt0cgmkm96t"
+            "ca1qdpy7jepxqyqfqghdnv0h7hyut4drfh2hyx9zef298l4u585wt0cgs7ysh3"
+
+        bech32KeyToAddrGolden
+            "ed25519_pk16cwyfh3jy6ls8ypfpen8pk55d3ufkp9793xc96u32rl83uuv89nsxsnnhg"
+            "ta1s0tpc3x7xgnt7qus9y8xvux6j3k83xcyhckymqhtj9g0u78n3sukwk7ku9u"
+            "ca1q0tpc3x7xgnt7qus9y8xvux6j3k83xcyhckymqhtj9g0u78n3sukwakffgx"
+
+bech32KeyToAddrGolden
+    :: Text
+    -- ^ public key ("ed25519_pk...")
+    -> Text
+    -- ^ corresponding testnet address ("ta...")
+    -> Text
+    -- ^ corresponding mainnet address ("ca...")
+    -> Spec
+bech32KeyToAddrGolden key testnetAddr mainnetAddr = it msg $ do
+    let xpub = XPub decodeBech32Key cc
+    encodeAddress (Proxy @(Jormungandr 'Testnet))
+        (singleAddressFromKey (Proxy @'Testnet) xpub) `shouldBe` testnetAddr
+    encodeAddress (Proxy @(Jormungandr 'Mainnet))
+        (singleAddressFromKey (Proxy @'Mainnet) xpub) `shouldBe` mainnetAddr
+
   where
-    cc = ChainCode "<ChainCode is not used by singleAddressToKey>"
+    msg = T.unpack key ++ " generates correct testnet and mainnet addresses"
+    decodeBech32Key :: ByteString
+    decodeBech32Key =
+        let
+            (hrp', dp) = either (error . show) id (Bech32.decodeLenient key)
+            hrp = T.unpack $ Bech32.humanReadablePartToText hrp'
+        in
+            if hrp == "ed25519_pk"
+            then maybe (error "invalid bech32") id (Bech32.dataPartToBytes dp)
+            else error $ "failed to decode key: " ++ hrp ++ " is not ed25519_pk"
+
+cc :: ChainCode
+cc = ChainCode "<ChainCode shouldn't be used by keyToAddress>"
```

Comment:
Note that these looks quite redundant with:

https://github.com/input-output-hk/cardano-wallet/blob/master/lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/CompatibilitySpec.hs#L50

Or am I missing something :thinking:  ?

Reply:
:s #435